### PR TITLE
Cleanup for slim header in page listing

### LINKED ISSUE #8864: UX Unification - Cleanup for slim header in page listing
### Issue Summary

#8792 introduced some issues due to conflict with other PRs that have been merged previously. In addition, there are also other improvements that can be made.

### Details

- [ ] When moving `.form-side` styles to its own `_form-side.scss` file, some updates introduced in #8671 weren't carried over. By the way, should this be in `components` rather than `layouts`?
  **Original:** https://github.com/wagtail/wagtail/blob/0693eed2454240b4d0f31da5166772bdd106dbb9/client/scss/layouts/_page-editor.scss#L61-L103
  **Moved:** https://github.com/wagtail/wagtail/blob/35b5ed350a30dfb626d55c0b2c021976c1f65761/client/scss/layouts/_form-side.scss#L5-L47
- [ ] When moving side panel init code to a separate `sidePanel.js` file, the original code in `page-editor.js` hasn't been removed. This results in the event listeners being registered twice, making the side panels inaccessible in the page editor (it immediately closes again). To fix this, remove the following: https://github.com/wagtail/wagtail/blob/35b5ed350a30dfb626d55c0b2c021976c1f65761/client/src/entrypoints/admin/page-editor.js#L333-L407
- [ ] The locale selector in the side panel should take the user to the localised page in the **page explorer**. Right now, it brings you to the editor instead. Possible solutions: add another keyword argument to `PageSidePanels`, e.g. `in_explorer`, or make a subclass e.g. `PageExplorerSidePanels` in case we need more customisations in the future.

There may be other issues I haven't noticed. Will update later if I find more.

### PR DESCRIPTION
**Pull request by laymonage**
Fixes #8864.

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by lb-** [comment]
@PaarthAgarwal can you read through this and check it all makes sense from your perspective

**Comment by lb-** [comment]
@laymonage thanks for resolving this. I'll try to review tonight

File: client/scss/layouts/_form-side.scss [hunk 1]
```diff
@@ -8,13 +8,12 @@
       w-right-0
```
Comment (laymonage) [inline client/scss/layouts/_form-side.scss:8]: Should this be in `components` instead of `layouts`? 🤔

**Review by thibaudcolas — APPROVED** [review]
Thank you for picking this up Sage 🙏 I had forgotten about the preview work’s form side panel changes, and totally missed we hadn’t removed the page-editor.js code.

For the "switch locales" link, I don’t think I agree with the rationale for the change. If the component looks the same in different places, to me it should also work the same. This is why for example we decided to have "Edit" under the Actions dropdown even on the Edit view. Let’s see what others think though!

File: client/scss/layouts/_form-side.scss (hunk 1 above)
Comment (thibaudcolas) [reply client/scss/layouts/_form-side.scss:8]: Indeed!

Comment (lb-) [reply client/scss/layouts/_form-side.scss:8]: Has been flagged on a future issue to fix up https://github.com/wagtail/wagtail/issues/8867

**Pull request merged**